### PR TITLE
chore(cdk): Update fact tables and tests for lambda layers v0.90.1

### PR DESCRIPTION
**Pull request by vasireddy99**
Add ADOT Lambda Layer ARNs for recent release

- https://aws-otel.github.io/docs/ReleaseBlogs/aws-distro-for-opentelemetry-lambda-layer-v0.36.0
- https://github.com/aws-observability/aws-otel-lambda/pull/818

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk-testing/framework-integ/test/aws-lambda/test/integ.lambda-adot.ts [hunk 1]
```diff
@@ -33,7 +33,7 @@ class StackUnderTest extends Stack {
       code: Code.fromInline(`exports.handler = ${handler.toString()}`),
       architecture: props.architecture,
       adotInstrumentation: {
-        layerVersion: AdotLayerVersion.fromJavaScriptSdkLayerVersion(AdotLambdaLayerJavaScriptSdkVersion.V1_17_1),
+        layerVersion: AdotLayerVersion.fromJavaScriptSdkLayerVersion(AdotLambdaLayerJavaScriptSdkVersion.V1_18_1),
```
Comment (lpizzinidev) [inline packages/@aws-cdk-testing/framework-integ/test/aws-lambda/test/integ.lambda-adot.ts:36]:
```suggestion
        layerVersion: AdotLayerVersion.fromJavaScriptSdkLayerVersion(AdotLambdaLayerJavaScriptSdkVersion.LATEST),
```
Can you please update all of these to use `LATEST`?

File: packages/aws-cdk-lib/aws-lambda/lib/adot-layers.ts [hunk 2]
```diff
@@ -398,7 +423,7 @@ export class AdotLambdaLayerGenericVersion extends AdotLambdaLayerVersion {
    * The latest layer version available in this CDK version. New versions could
    * introduce incompatible changes. Make sure to test them before deploying to production.
    */
-  public static readonly LATEST = this.V0_88_0;
+  public static readonly LATEST = this.V0_90_1;
   private constructor(protected readonly layerVersion: string) {
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-lambda/lib/adot-layers.ts:427]:
```suggestion
  public static readonly LATEST = this.V0_90_1;

  private constructor(protected readonly layerVersion: string) {
```
Spacing

File: packages/aws-cdk-lib/aws-lambda/test/adot.test.ts [hunk 3]
```diff
@@ -21,18 +21,18 @@ describe('ADOT Lambda Layer', () => {
     });
 
     test('is added properly when the region information is available at synthesis time', () => {
-      const layerArn = lambda.AdotLambdaLayerJavaSdkVersion.V1_31_0.layerArn(fn.stack, fn.architecture);
+      const layerArn = lambda.AdotLambdaLayerJavaSdkVersion.V1_32_0.layerArn(fn.stack, fn.architecture);
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-lambda/test/adot.test.ts:24]:
```suggestion
      const layerArn = lambda.AdotLambdaLayerJavaSdkVersion.LATEST.layerArn(fn.stack, fn.architecture);
```
Same as with integration, we could use `LATEST` for the unit tests as well.

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Thanks 👍
I left some notes for minor adjustments on tests.

File: packages/aws-cdk-lib/aws-lambda/test/adot.test.ts (hunk 3 above)
Comment (vasireddy99) [reply packages/aws-cdk-lib/aws-lambda/test/adot.test.ts:24]: This test is specific to get with the version tag, there is already a test for the latest tag. so the changes are not required here. but the rest of the places are changed to LATEST.

**Review by lpizzinidev — APPROVED** [review]
Thanks for the changes and the clarification 👍

**Review by paulhcsun — APPROVED** [review]
Thanks for adding this @vasireddy99, and thank you for reviewing @lpizzinidev!

**Pull request merged**